core/trino-parser/src/main/java/io/trino/sql/tree/Measure.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class Measure
```

Comment:
Instead of a dedicated Measure class, it would be more appropriate to have something like `Over`/`WindowedOperation` that takes an `Expression` and a `Window` clause. The `Expression` could be a FunctionCall or an Identifier, for the function call vs measure case, respectively.

Reply:
I agree that it would be a very good change, but it mostly comes down to refactoring the `FunctionCall`.
Currently, a `FunctionCall` carries an optional `Window` through the Optimizer, while every `Window` should be planned as a `WindowNode`, and never get past the `QueryPlanner`. 
Extracting `Window` from `FunctionCall` is not quite straightforward. It affects the analysis a lot, and IIRC, there are plan tests using a `FunctionCall` for building framed window functions.

I suggest that we have the `Measure` class for now, and as a follow-up we can refactor `FunctionCall`. Then `Measure` can become `WindowOperation` and capture both.

Reply:
I'm ok with that, but I would go ahead and call that new class `WindowOperation` and make a note that it only handles measures for now but should be retrofitted to handle function calls as well.